File: src/main/resources/templates/admin/assets/js/model/dto/MemberDTO.js
import {MemberPackageDTO} from "/templates/admin/assets/js/model/dto/MemberPackageDTO.js";
import {UserDTO} from "/templates/admin/assets/js/model/dto/UserDTO.js";

export class MemberDTO {
    /**
     * Constructor for MemberDTO
     * @param {UserDTO} userEntity
     * @param {MemberPackageDTO} memberPackageEntity
     * @param memberPackageEntityPackageEntityId
     * @param memberPackageEntityStartDate
     * @param sortFieldName
     * @param sortDirection
     */
    constructor(userEntity, memberPackageEntity, memberPackageEntityPackageEntityId, memberPackageEntityStartDate, sortFieldName, sortDirection) {
        this.userEntity = userEntity;
        this.memberPackageEntity = memberPackageEntity;
        this.memberPackageEntityPackageEntityId = memberPackageEntityPackageEntityId;
        this.memberPackageEntityStartDate = memberPackageEntityStartDate;
        this.sortFieldName = sortFieldName;
        this.sortDirection = sortDirection;
    }
}

File: src/main/java/com/web/model/dto/MemberDTO.java
package com.web.model.dto;

import com.web.enums.operation.SortDirection;
import jakarta.validation.Valid;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MemberDTO {
    private Long id;
    @Valid
    private UserDTO userEntity;
    private String occupation;
    private String hobby;
    private String note;
    private String memberPackageEntityPackageEntityId;
    private Date memberPackageEntityStartDate;
    private String sortFieldName;
    private SortDirection sortDirection;
}
